modules/javafx.graphics/src/main/native-glass/gtk/glass_key.cpp

```diff
@@ -284,6 +284,63 @@ gint find_gdk_keyval_for_glass_keycode(jint code) {
     return result;
 }
 
+static int search_keys(GdkKeymapKey* keys, gint n_keys, int search_group)
+{
+    int result = -1;
+    for (gint i = 0; i < n_keys; ++i)
+    {
+        if (keys[i].group == search_group &&
+            keys[i].level == 0)
+        {
+            result = keys[i].keycode;
+            break;
+        }
+    }
+    return result;
+}
+
+extern "C" {
+    static int get_current_keyboard_group();
+}
+
+int find_gdk_keycode_for_keyval(gint keyval)
+{
+    Display *xdisplay = gdk_x11_get_default_xdisplay();
+    GdkKeymapKey *keys = nullptr;
+    gint n_keys = 0;
+    GdkKeymap* keymap = gdk_keymap_get_default();
```

Comment:
` gdk_keymap_get_for_display(gdk_display_get_default())` is the non deprecated way.